tracing-subscriber/src/layer.rs

```diff
@@ -586,9 +586,8 @@ pub struct Identity {
 /// [`Context::scope`]: struct.Context.html#method.scope
 #[cfg(feature = "registry")]
 #[cfg_attr(docsrs, doc(cfg(feature = "registry")))]
-pub struct Scope<'a, L: LookupSpan<'a>>(
-    Option<std::iter::Chain<registry::FromRoot<'a, L>, std::iter::Once<SpanRef<'a, L>>>>,
-);
+#[deprecated(note = "moved to crate::registry::ScopeFromRoot")]
+pub type Scope<'a, L> = std::iter::Flatten<std::option::IntoIter<registry::ScopeFromRoot<'a, L>>>;
```

Comment:
> note that if the type was previously an alias rather than a newtype, this would be a breaking change

Yeah, my thought here was mostly that this'd be committing to the deprecation and replacement, and that there'd be a whole bunch of newtype overhead otherwise.

But I'd be fine with reverting this to newtypes if you prefer that.

Reply:
Do you also want to turn `registry::{Parents,FromRoot}` back into newtypes? Considering that both of those correspond 1:1 with `registry::{Scope,ScopeFromRoot}`.

Reply:
Did at least newtype `layer::Scope` now.

Reply:
Converted the others to newtypes as well.